types/src/bn254_circom.rs

```diff
@@ -0,0 +1,269 @@
+    )
+    .unwrap();
+
+    let beta_g2 = g2_projective_str_to_affine(
+        [
+            "1125365732643211423779651913319958385653115422366520671538751860820509133538",
+            "10055196097002324305342942912758079446356594743098794928675544207400347950287",
+        ],
+        [
+            "10879716754714953827605171295191459580695363989155343984818520267224463075503",
+            "440220374146936557739765173414663598678359360031905981547938788314460390904",
+        ],
+    )
+    .unwrap();
+
+    let gamma_g2 = g2_projective_str_to_affine(
+        [
+            "10857046999023057135944570762232829481370756359578518086990519993285655852781",
+            "11559732032986387107991004021392285783925812861821192530917403151452391805634",
+        ],
+        [
+            "8495653923123431417604973247489272438418190587263600148770280649306958101930",
+            "4082367875863433681332203403145435568316851327593401208105741076214120093531",
+        ],
+    )
+    .unwrap();
+
+    let delta_g2 = g2_projective_str_to_affine(
+        [
+            "19799867077440075892798570892827678991452882191483986973420950266983588147526",
+            "7261406229996412667156189606964369006242293247396567701023787052439810543589",
+        ],
+        [
+            "15618356441847575237880159451782511420373837463064250522093342825487687558812",
+            "20490123502151072560031041764173142979409281632225526952209676367033524880945",
+        ],
+    )
+    .unwrap();
+
+    let mut gamma_abc_g1 = Vec::new();
+    for points in [
+        g1_projective_str_to_affine(
+            "16119992548622948701752093197035559180088659648245261797962160821523395857787",
+            "10895012769720065848112628781322097989082134121307195027616506940584635557433",
+        )
+        .unwrap(),
+        g1_projective_str_to_affine(
+            "12743680909720798417558674763081930985009983383780261525309863653205478749832",
+            "10808093222645961212778297519773755506856954740368509958745099866520706196565",
+        )
+        .unwrap(),
+    ] {
+        gamma_abc_g1.push(points);
+    }
+
+    let vk = VerifyingKey {
+        alpha_g1,
+        beta_g2,
+        gamma_g2,
+        delta_g2,
+        gamma_abc_g1,
+    };
+
+    PreparedVerifyingKey::from(vk)
+}
+
+fn parse_field_element(s: &str) -> Result<Fq, CryptoMaterialError> {
+    s.parse::<Fq>()
+        .map_err(|_e| CryptoMaterialError::DeserializationError)
+}
+
+macro_rules! serialize {
+    ($obj:expr, $method:ident) => {{
+        let mut buf = vec![];
+        $obj.$method(&mut buf)?;
+        buf
+    }};
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Hash, Serialize)]
+pub struct Bn254G1ProjectiveCompressed(pub(crate) [u8; G1_PROJECTIVE_COMPRESSED_NUM_BYTES]);
+
+impl Bn254G1ProjectiveCompressed {
+    pub fn new(x: &str, y: &str) -> anyhow::Result<Self> {
+        let g1 = ark_bn254::G1Projective::new_unchecked(
```

Comment:
Note to self: Using `_unchecked` sounds like a problem.

Reply:
it does, but checked uses assert! and can potentially panic.  it's annoying.  We could call the checks after (would need to convert to affine).

But we need to deserialize to affine eventually anyway which has the checks.  but yeah not ideal